snuba/admin/views.py

```diff
@@ -441,8 +442,58 @@ def clickhouse_trace_query() -> Response:
         )
 
     try:
-        result = run_query_and_get_trace(storage, raw_sql)
-        return make_response(jsonify(asdict(result)), 200)
+        if not raw_sql.endswith("SETTINGS log_profile_events=1"):
```

Comment:
You will have to handle the case where the input query already has a settings clause. E.g.

```
SELECT ...
FROM ...
SETTINGS max_threads=1
```

Reply:
Done.